.github/workflows/codeql-analysis.yml

```diff
@@ -21,12 +28,12 @@ jobs:
 
     steps:
     - name: Checkout repository
-      uses: actions/checkout@v2
+      uses: actions/checkout@629c2de402a417ea7690ca6ce3f33229e27606a5 # v2
```

Comment:
@oh6hay hardcoding a specific commit means it will not pull the latest CodeQL queries added. We are currently leveraging this and any new CodeQL queries committed to the CodeQL repo helps us identify issues on existing codebase. I am afraid we will lose this advantage if we specify a specific commit.

Reply:
that's right